testworld/signature_test.go

```diff
@@ -0,0 +1,115 @@
+// +build testworld
+
+package testworld
+
+import (
+	"github.com/centrifuge/centrifuge-protobufs/gen/go/coredocument"
+	"github.com/centrifuge/go-centrifuge/config"
+	"github.com/centrifuge/go-centrifuge/crypto"
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/documents/purchaseorder"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/testingutils/config"
+	"github.com/centrifuge/go-centrifuge/testingutils/documents"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"testing"
+	"time"
+)
+
+func TestHost_ValidSignature(t *testing.T) {
+	t.Parallel()
+
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+
+	ctxh := testingconfig.CreateAccountContext(t, alice.host.config)
+
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, alice.id, alice.host.config)
+	assert.Equal(t, 1, len(dm.Signatures()))
+
+	signatures, _, _ := alice.host.p2pClient.GetSignaturesForDocument(ctxh, dm)
+	assert.Equal(t, 1, len(signatures))
+}
+
+func TestHost_FakedSignature(t *testing.T) {
+	t.Parallel()
+
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	ctxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, eve.id, alice.host.config)
+
+	signatures, signatureErrors, _ := eve.host.p2pClient.GetSignaturesForDocument(ctxh, dm)
+	assert.Error(t, signatureErrors[0], "Signature verification failed error")
+	assert.Equal(t, 0, len(signatures))
+}
+
+func TestHost_RevokedSigningKey(t *testing.T) {
```

Comment:
LGTM 
But lets wait until we implement the revoked key validation to check this in.